pkg/upstream/cluster/edf.go

```diff
@@ -54,16 +58,27 @@ func (edf *edfSchduler) NextAndPush(weightFunc func(item WeightItem) float64) in
 	entry := heap.Pop(&edf.items).(*edfEntry)
 	edf.currentTime = entry.deadline
 	weight := weightFunc(entry.item)
-	// update the deadline and put into priorityQueue again
+	// update the index、deadline and put into priorityQueue again
+	edf.globalIndex++
```

Comment:
maybe overflow ?

Reply:
I think it is difficult for int64 to overflow.

Reply:
or reset 0 when overflow?

Reply:
if reset 0, the result will be wrong.
use time.Now()，and compare time？